modules/offload-s3/module.go

```diff
@@ -217,16 +214,23 @@ func (m *Module) RootHandler() http.Handler {
 	return nil
 }
 
-func (m *Module) list(ctx context.Context) error {
+func (m *Module) VerifyBucket(ctx context.Context) error {
+	if m.BucketExists {
```

Comment:
`VerifyBucket` may need to be atomic

Reply:
in general it's not meant to be called concurrently, even if it is just a way to decrease the number of requests to s3